ADS-2059: Shopify marketing consent and GDPR documentation (#33)
supercid/wiki-docs: Marketing-Permission-And-Gdpr-Compatibility.md

> Marketing permission or newsletter constent from customers are recorded at Nosto either during order confirmation (Conversion Tracking) or through customerUpdate webhook (Customer Feed). In case of order confirmation, the consent is received through the `marketingPermission` field in `OrderCustomer` bean and though `acceptsMarketing` field in `ShopifyCustomer` bean in case of webhook. These fields tells Nosto whether the customer has given their [GDPR](https://www.eugdpr.org/) compliant consent for receiving marketing emails.

Reviewer: ```suggestion
Marketing permission or newsletter consent from customers is recorded either during order confirmation or through customer webhooks. In case of order confirmation, the consent is received through the `marketingPermission` flag. These fields tell Nosto if the customer has given their [GDPR](https://www.eugdpr.org/) compliant consent for receiving marketing emails.
```

1. We currently handle `customers/create` and `customers/update` webhooks. I'd leave it general as customer webhook so we don't need to keep updating the documentation when we handle other webhook types.
2.  The `OrderCustomer` bean is internal. No point in documenting this externally.
3. You could link to https://docs.nosto.com/techdocs/implementing-nosto/implement-on-your-website/manual-implementation/adding-the-customer-information

![Consent during order confirmation](https://user-images.githubusercontent.com/82023195/136540098-3719ca4f-77e6-47d1-8dfa-525010885064.png)

When the customer selects `Email me with news and offers` option and confims the order, the consent along with order information will be sent to Nosto through `/order/track` API in the following format,

```json
{
    "info": {
        "order_number":"4321633632470",
        "type":"order",
        "email":"manikandan.ravikumar@nosto.com",
        "first_name":"Mani",
        "last_name":"Nosto", 
        "newsletter": "true"
    },
    "items": [{
        "product_id":"7190446571734",
        "sku_id":"41479723778262",
        "name":"Cool Kicks",
        "quantity":1,
        "unit_price":123.12,
        "price_currency_code":"EUR"
    }]
}
```

## Consent - Customer update
A merchant will be able to update customer's information or their marketing constent from store management (Admin) page. 

![Update Marketing Consent](https://user-images.githubusercontent.com/82023195/136540161-057d6c72-fa25-4785-aa56-19a48a016200.png)

> During this process, shopify will share updated customer information through a registered webhook `https://webhooks.nosto.com/hub/shopify/customerFeed` in the following format,

Reviewer: ```suggestion
During this process, shopify will share updated customer information through a registered webhook `https://webhooks.nosto.com/hub/shopify/customerFeed` in the following format,
```

The URL of our webhook and payload are internal implementation details. I wouldn't go into that much detail in our documentation. I think it's sufficient to simply say that we handle marketing permission changes automatically when we receive webhooks from Shopify.

Author: Agreed

```json
{
    "id": "12345",
    "email":"manikandan.ravikumar@nosto.com",
    "first_name":"Mani",
    "last_name":"Nosto", 
    "accepts_marketing": true
}
```

### Note: 
1. If your store has a custom process for gathering marketing permissions or the default logic explained above doesn't fit for your store, you can override Nosto's extension to amend the behaviour. Please read more detailed instructions in our guide on [Import customer data](https://help.nosto.com/en/articles/2884483-how-to-import-customer-data-to-nosto-via-api)

> 2. In order manually override marketing information for a customer email, follow the steps here [Manually Override Consent](https://docs.nosto.com/techdocs/apis/rest/customers/toggling-email-opt-in-using-the-consent-api)

Reviewer: You could also link to the tagging guide here https://docs.nosto.com/techdocs/implementing-nosto/implement-on-your-website/manual-implementation/adding-the-customer-information

Author: Done
